lib/Importer/Caffe2.cpp

```diff
@@ -314,9 +314,16 @@ void caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
       auto wDims = flattenCdr(w->dims(), axis_w);
       Tensor tmp(ElemKind::FloatTy, {wDims.first, wDims.second});
       tmp.copyRawFrom(w);
-      tmp.transpose(&wtag, {1, 0});
-    } else
+      if (typeName == "FC") {
```

Comment:
This feels a bit unnecessary, with two separate checks for `if (typeName == "FC")`. Is it possible to wrap all the logic already in here for transposing weights in `if (typeName == "FC")`, and else just assign `wtag`?

Reply:
We need to flatten the w if its dim size is larger than 2.  So here is the logic:
1) if w's dim size > 2 and FC  then flatten and transpose.
2) if w's dim size <= 2 and FC then transpose.
3) if w's dim size > 2 and FCT then flatten and assign.
4) if w's dim size <= 2 and FCT then assign. 
Now to reused the "flatten" code, I use w'sdim size >2 as the outer condition. If we could like to use the (typename == "FC" )  as the outer condition, I think we have to duplicate the "flatten" code.

Reply:
Got it. Would it work to perform flatten if necessary, and then perform transpose if necessary? Something like:

```cpp
if (w->dims().size() > 2) {
  // flatten logic
}

if (typeName == "FC") {
  // transpose logic
}
```

Reply:
Yes, we could do that. But in that case, no mater the flatten is needed, we need to copy w to a temp tensor(directly or flatten) and then do the transpose or direct copy.  I remember we added " w->dims().size() > 2" to avoid non-necessary copy. Otherwise, we can simply call "flatten" and then  run the tranpose logic.

Reply:
Ah ok, got it. In that case I wonder if we could use `std::move()` here instead of `assign()` to avoid any unnecessary copies? E.g. 

```cpp
if (w->dims().size() > 2) {
  // tmp = flattened w
  wtag = std::move(tmp);
} else {
  wtag = std::move(*w);
}
```

Reply:
True! If two FCs in the proto reference use the same weight then it would be an issue. I suppose such a network is technically valid, though unlikely in practice and not yet seen. BTW, we are currently using `std::move(*b)` for creating the bias (`Variable B`). So that should also be fixed for the same reason.

Anyway, this means no matter what we need to make an extra copy of `w`. So in the snippet I wrote above in the else case, if we replace `wtag = std::move(*w);` with `wtag.assign(w);` I believe it works fine, correct?

Reply:
Yes. In my last version,  std::move is applied to Tensor tmp as you suggested, but keep using wtag.assign(w) in the else part.